From a449027cf5aaa611cd431fe2b7cca1e8d31e34f1 Mon Sep 17 00:00:00 2001
From: Alexey Edelev <alexey.edelev@qt.io>
Date: Tue, 13 May 2025 14:32:57 +0200
Subject: [PATCH 108/553] Introduce _qt_internal_android_get_package_source_dir

Function returns the custom target QT_ANDROID_PACKAGE_SOURCE_DIR
which contains templates for the Android deployment.

The function has the scope constraint, which only allows calling
it inside Android finalizer.

Pick-to: 6.8 6.9 6.10.0
Change-Id: Ia8fb9751120f9995700b03ddb5a1b8fb3a568e92
Reviewed-by: Assam Boudjelthia <assam.boudjelthia@qt.io>
(cherry picked from commit 403f04b571635173a5c4eef4db95f31bc0113673)
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
---
 src/corelib/Qt6AndroidMacros.cmake | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/src/corelib/Qt6AndroidMacros.cmake b/src/corelib/Qt6AndroidMacros.cmake
index 503b217c425..2ffd31d2a3e 100644
--- a/src/corelib/Qt6AndroidMacros.cmake
+++ b/src/corelib/Qt6AndroidMacros.cmake
@@ -1799,7 +1799,7 @@ function(_qt_internal_android_get_target_android_build_dir out_build_dir target)
 endfunction()
 
 function(_qt_internal_expose_android_package_source_dir_to_ide target)
-    get_target_property(android_package_source_dir ${target} QT_ANDROID_PACKAGE_SOURCE_DIR)
+    _qt_internal_android_get_package_source_dir(android_package_source_dir ${target})
     if(android_package_source_dir)
         get_target_property(target_source_dir ${target} SOURCE_DIR)
         if(NOT IS_ABSOLUTE "${android_package_source_dir}")
@@ -1942,6 +1942,17 @@ function(_qt_internal_android_find_asan_wrap_sh out_wrap_sh_path)
     endif()
 endfunction()
 
+# Return the path to the target template directory if it's set for the target.
+# Then this path is stored in the target QT_ANDROID_PACKAGE_SOURCE_DIR property
+# and can only be effectively read in android finalizers.
+function(_qt_internal_android_get_package_source_dir out_var target)
+    get_target_property(package_src_dir ${target} QT_ANDROID_PACKAGE_SOURCE_DIR)
+    if(NOT package_src_dir)
+        set(package_src_dir "")
+    endif()
+    set(${out_var} "${package_src_dir}" PARENT_SCOPE)
+endfunction()
+
 # Returns the path to the Android platform-tools(adb is located there).
 function(_qt_internal_android_get_platform_tools_path out_var)
     set(${out_var} "${ANDROID_SDK_ROOT}/platform-tools" PARENT_SCOPE)
-- 
2.51.2

